packages/block-logs-stream/src/fetchLogs.ts

```diff
@@ -158,6 +160,18 @@ export async function* fetchLogs<abiEvents extends readonly AbiEvent[]>({
         continue;
       }
 
+      if (
+        retryCount < maxRetryCount &&
+        (MISSING_BLOCK_ERRORS.some((e) => error.message.includes(e)) ||
+          error.message === new BlockNotFoundError({ blockNumber: toBlock }).message)
+      ) {
+        const seconds = 2 * retryCount;
+        debug(`missing block, retrying in ${seconds}s`);
+        await wait(1000 * seconds);
```

Comment:
the main reason we'd run into this is a load balanced RPC, in which case much less than 2s wait time should be enough for the RPCs to be back in sync. Should we do the first retry after like 500ms?

Reply:
the first retry happens immediately (retryCount starts at zero)